app/controllers/candidate_interface/english_foreign_language/review_controller.rb

```diff
@@ -1,8 +1,45 @@
 module CandidateInterface
   module EnglishForeignLanguage
     class ReviewController < CandidateInterfaceController
+      before_action :check_for_english_language_proficiency
+
       def show
-        @english_language_proficiency = current_application.english_language_proficiency
+        @component_instance = derive_component_instance(english_language_proficiency)
```

Comment:
Interesting. Could this logic live in the view (another component perhaps) instead of the controller, to slim down our controller and make this easier to unit test?

Reply:
I was thinking I'd extract this into its own model (or service object, probably) when I add the 'no qualification' review page.

Reply:
@malcolmbaig 👌